### ISSUE TITLE
Feature request: valid JSON save files

### ISSUE THREAD

**Issue by UndeRus** [yairm210/Unciv#9414]
I wrote multiplayer unciv server and found that Unciv save files contains broken json **https://javadoc.io/static/com.badlogicgames.gdx/gdx/1.11.0/com/badlogic/gdx/utils/JsonWriter.OutputType.html#minimal**, which can't be ready by something except libgdx. Can be this fixed or there is a reason why it is done?

**Comment by UndeRus** [comment]
My first idea was to create server with server side replay tool with more efficient save files storage, e.g. msgpack and storing turn diffs instead full saves, but with current json format it can be achieved only with JVM languages.

**Comment by yairm210** [comment]
I know that in the [server](https://github.com/touhidurrr/UncivServer.xyz) they manage to parse this json and there they use node.js, maybe that can help you out
As for us, we're not planning on changing our save format, but we are open to retaining conversion to regular json - this sounds like a language-specific thing though

**Comment by CrsiX** [comment]
Since this issue is related to multiplayer behavior, just a side note: I'm always uploading _valid_ JSON for APIv2 servers, since parsing them successfully is crucial for just any meaningful server-side logic.

> Is there any reason _not_ to use the slightly space saving variant?

Yes. Interoperability. If you want to save space, compress it. With gzip I'm reaching about 10% to 20% of space (depending on game), with lz4 and brotli even a bit more (but not planned to be used, gzip is just fine).

> My first idea was to create server with server side replay tool with more efficient save files storage, e.g. msgpack and storing turn diffs instead full saves, but with current json format it can be achieved only with JVM languages.

Interesting. Keep me updated about that, please. As far as I can tell, implementing diff files is more complicated than I first thought. It would need a lot of refactoring in the client, probably. That's why I postponed that idea already. Or, do you mean to just save diffs on the server for replay, so that the clients don't need to interact with those diff files at all?

TL;DR: I agree with having valid JSON as default output format. Minimal output may be of some use to save disk space, but compressing it already handles all that "bloat" of the valid JSON structures.
